sentry_sdk/integrations/stdlib.py

```diff
@@ -113,28 +136,123 @@ def putrequest(self, method, url, *args, **kwargs):
-            return real_getresponse(self, *args, **kwargs)
-
         rv = real_getresponse(self, *args, **kwargs)
 
-        span.set_http_status(int(rv.status))
-        span.set_data("reason", rv.reason)
-        span.finish()
+        if span is not None:
+            span.set_http_status(int(rv.status))
+            span.set_data("reason", rv.reason)
+            span.finish()
+
+        url = getattr(self, "_sentrysdk_url", None)  # type: Optional[str]
+        if url is None:
+            return rv
+
+        response_body = None
+        if getattr(self, "_sentrysdk_is_graphql_request", False):
+            with capture_internal_exceptions():
+                response_data = rv.read()
+                # once we've read() the body it can't be read() again by the
+                # app; save it so that it can be accessed again
+                rv.read = io.BytesIO(response_data).read
+                try:
+                    # py3.6+ json.loads() can deal with bytes out of the box, but
+                    # for older version we have to explicitly decode first
+                    response_body = json.loads(response_data.decode())
+                except (JSONDecodeError, UnicodeDecodeError, TypeError):
+                    return rv
+
+        if isinstance(response_body, dict) and response_body.get("errors"):
+            method = getattr(self, "_sentrysdk_method", None)  # type: Optional[str]
+            request_body = getattr(self, "_sentry_request_body", None)
+            hub = Hub.current
+            with hub.configure_scope() as scope:
+                scope.add_event_processor(
+                    _make_request_processor(
+                        url, method, rv.status, request_body, response_body
+                    )
+                )
+                try:
+                    raise SentryGraphQLClientError
+                except SentryGraphQLClientError as ex:
+                    event, hint = event_from_exception(
+                        ex,
+                        client_options=hub.client.options if hub.client else None,
+                        mechanism={
+                            "type": StdlibIntegration.identifier,
+                            "handled": False,
+                        },
+                    )
+
+            hub.capture_event(event, hint=hint)
 
         return rv
 
     HTTPConnection.putrequest = putrequest
+    HTTPConnection.send = send
     HTTPConnection.getresponse = getresponse
 
 
+def _make_request_processor(url, method, status, request_body, response_body):
+    # type: (str, Optional[str], int, Any, Any) -> EventProcessor
+    def stdlib_processor(
+        event,  # type: Dict[str, Any]
+        hint,  # type: Dict[str, Tuple[type, BaseException, Any]]
+    ):
+        # type: (...) -> Optional[Event]
+        with capture_internal_exceptions():
+            request_info = event.setdefault("request", {})
+
+            parsed_url = parse_url(url, sanitize=False)
```

Comment:
It's always the same endpoint, but it can be queried either with GET or POST. If it's POST, all data is in the request body, if it's GET, it'll be in the query params. I'm guessing you could have personal data in the query string. Should probably wrap that in `_should_send_default_pii()`, too.

Reply:
I also added the guard to saving the `query_string`.